### PR TITLE
Adapt to the new ChannelFactory/ServerChannelFactory

### PR DESCRIPTION
**Pull request by violetagg**
In Netty 5 we have now

```
/**
 * Creates a new {@link Channel}.
 */
public interface ChannelFactory<T extends Channel> {
    /**
     * Creates a new channel.
     */
    T newChannel(EventLoop eventLoop) throws Exception;
}
```

```
/**
 * Creates a new {@link ServerChannel}.
 */
public interface ServerChannelFactory<T extends ServerChannel> {
    /**
     * Creates a new channel.
     */
    T newChannel(EventLoop eventLoop, EventLoopGroup childEventLoopGroup) throws Exception;
}
```

Related to #1873

### CODE REVIEW

File: reactor-netty-core/src/main/java/reactor/netty/transport/ServerTransportConfig.java [hunk 1]
```diff
@@ -161,7 +160,12 @@ protected ServerTransportConfig(ServerTransportConfig<CONF> parent) {
...
 	@Override
 	protected Class<? extends Channel> channelType(boolean isDomainSocket) {
-		return isDomainSocket ? ServerDomainSocketChannel.class : ServerSocketChannel.class;
+		throw new UnsupportedOperationException();
+	}
+
+	@Override
+	protected ChannelFactory<? extends Channel> connectionFactory(boolean isDomainSocket) {
+		throw new UnsupportedOperationException();
 	}
```
Comment (pderop) [inline reactor-netty-core/src/main/java/reactor/netty/transport/ServerTransportConfig.java:169]: are these UnsupportedException temporary ? I mean: is there a TODO in order to fix these UnsupportedOperationException ?

Comment (violetagg) [reply reactor-netty-core/src/main/java/reactor/netty/transport/ServerTransportConfig.java:169]: No `ServerTransportConfig` cannot support `SocketChannel`. See `channelType` javadoc.

Comment (pderop) [reply reactor-netty-core/src/main/java/reactor/netty/transport/ServerTransportConfig.java:169]: ok.

**Review by pderop — APPROVED** [review]
LGTM

**Comment by violetagg** [comment]
@pderop Thanks for the review

**Pull request merged**